### PR TITLE
docs: Introduce quick start page

### PR DESCRIPTION
**Pull request by ppawlowski**
## Description

This pull request introduces a "Quick Start" page which aims to guide self-hosted customers on how to spin up the FlowFuse platform in the shortest possible time.

## Related Issue(s)

Should be introduced in parallel with https://github.com/FlowFuse/docker-compose/pull/158
Closes https://github.com/FlowFuse/flowfuse/issues/4631

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: docs/quick-start/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+1. A domain name that you own and can configure DNS settings for (explained in step 1)
+2. Docker and Docker Compose installed on your system (either as a standalone binary or as docker plugin)
+
+## Step 1: Configure DNS
+
+Before running the FlowFuse, you need to configure your fully qualified domain name settings:
+
+1. Point your main domain (e.g., `yourdomain.com`) to your server's IP address (works with subdomain as well).
+2. Set up a wildcard DNS record (e.g., `*.yourdomain.com`) to point to your server's IP address.
```
Comment (hardillb) [inline docs/quick-start/README.md:34]: Should this explicitly mention an A record

File: docs/quick-start/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+Before you begin, ensure you have the following:
+
+1. A domain name that you own and can configure DNS settings for (explained in step 1)
+2. Docker and Docker Compose installed on your system (either as a standalone binary or as docker plugin)
+
+## Step 1: Configure DNS
+
+Before running the FlowFuse, you need to configure your fully qualified domain name settings:
+
+1. Point your main domain (e.g., `yourdomain.com`) to your server's IP address (works with subdomain as well).
```
Comment (hardillb) [inline docs/quick-start/README.md:33]:
```suggestion
1. Point your main domain (e.g., `example.com`) to your server's IP address (this works with subdomain as well e.g. `flowfuse.example.com).
```

File: docs/quick-start/README.md (hunk 1 above)
Comment (hardillb) [inline docs/quick-start/README.md:34]:
```suggestion
2. Set up a wildcard DNS record (e.g., `*.example.com`) to point to your server's IP address.
```

File: docs/quick-start/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+```bash
+curl -o docker-compose.yml https://raw.githubusercontent.com/FlowFuse/docker-compose/refs/heads/main/docker-compose-quick-start.yml
+```
+
+## Step 3: Start the Application
+
+Run the following command to deploy FlowFuse. Replace the  `yourdomain.com` with your domain name configured in step 1.:
+
+```bash
+DOMAIN=yourdomain.com docker compose -p flowfuse up -d
```
Comment (hardillb) [inline docs/quick-start/README.md:49]:
```suggestion
DOMAIN=example.com docker compose -p flowfuse up -d
```

File: docs/quick-start/README.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+```
+
+This command will download the necessary Docker images, run initial setup and start all the required services in detached mode.
+
+## Step 4: Complete the application Setup
+
+Open your web browser and navigate to `http://forge.yourdomain.com` . You will be redirected to the setup page where you can create your admin account and set up your instance.
```
Comment (hardillb) [inline docs/quick-start/README.md:56]:
```suggestion
Open your web browser and navigate to `http://forge.example.com/setup` . You will be redirected to the setup page where you can create your admin account and set up your instance.
```

File: docs/quick-start/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+For detailed information about first setup and configuration, please follow [this guide](../install/first-run.md).
+
+
+## Cleanup
+
+To stop and remove the FlowFuse application, run the following command. Replace `yourdomain.com` with your domain name:
+
+```bash
+DOMAIN=yourdomain.com docker compose -p flowfuse down -v
```
Comment (hardillb) [inline docs/quick-start/README.md:65]:
```suggestion
DOMAIN=example.com docker compose -p flowfuse down -v
```

**Review by hardillb — CHANGES_REQUESTED** [review]
I have made some suggestions

I also think we should include the `-p` project option so the default directory name is not used.

Also need to think about upgrade instructions.

File: docs/quick-start/README.md (hunk 1 above)
Comment (ppawlowski) [reply docs/quick-start/README.md]: > Should this explicitly mention an A record

Added.

**Comment by ppawlowski** [comment]
> we should include the `-p` project option so the default directory name is not used.

Actually, I forgot to remove it, since it is covered by the [compose file](https://github.com/FlowFuse/docker-compose/pull/158/files#diff-16f23a4d5dac0481e4a6777f3a9ffddda444ac4d0701f68eccea0d780abcb222R1), with `name` key. I would like to keep it as simple as possible.
@hardillb let me know what you think.

**Comment by ppawlowski** [comment]
> Also need to think about upgrade instructions.

This is a quick-start page. The goal is to quickly start using FlowFuse. If someone wants to go deeper with FlowFuse on Docker, this is not the page for them - they should go to https://flowfuse.com/docs/install/docker/ , which quick-start page mentions at the end.

**Comment by hardillb** [comment]
hmm, OK.

In that case there should also be a note saying that the quick start doesn't support HTTPS

**Comment by ppawlowski** [comment]
>  there should also be a note saying that the quick start doesn't support HTTPS

I'm against listing what this quick-start compose file doesn't do. I'm worried, that this list will grow each time someone reports that some other configuration is missing.
Instead, in the page intro we could list what it will do (create FF components, create a database, etc) and link to the 'docker install' page for advanced configuration.

**Comment by joepavitt** [comment]
Making some minor formatting & grammar changes. Also want to consider where we link to this new "Quick Start" guide.

**Comment by joepavitt** [comment]
Spoken with Piotr, we've agreed to add the "Checklist" on the Docker Install and k8s install pages, with recommendations on what needs to be configured for production environments. One minor amendment incoming which clarifies this Quick Start docs is _not_ for production.

**Review by joepavitt — APPROVED** [review]

**Comment by joepavitt** [comment]
All good as far as I'm concerned, just need @hardillb to unblock via his requested changes review

**Review by hardillb — APPROVED** [review]

**Comment by hardillb** [comment]
All good, merge when ever

**Pull request merged**